ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java

```diff
@@ -138,40 +183,87 @@ private CacheConfiguration createCacheConfiguration(AssociationKeyMetadata assoc
+		cacheConfiguration.setAtomicityMode( CacheAtomicityMode.TRANSACTIONAL );
 
 		QueryEntity queryEntity = new QueryEntity();
-		queryEntity.setKeyType( getEntityIdClassName( entityKeyMetadata.getTable(), context ) );
+
+		queryEntity.setTableName( entityKeyMetadata.getTable() );
+		queryEntity.setKeyType( getEntityIdClassName( entityKeyMetadata.getTable(), context ).getSimpleName() );
 		queryEntity.setValueType( StringHelper.stringAfterPoint( entityKeyMetadata.getTable() ) );
+		addTableInfo( queryEntity, context, entityKeyMetadata.getTable() );
+
 		for ( AssociationKeyMetadata associationKeyMetadata : context.getAllAssociationKeyMetadata() ) {
 			if ( associationKeyMetadata.getAssociationKind() != AssociationKind.EMBEDDED_COLLECTION
 					&& associationKeyMetadata.getTable().equals( entityKeyMetadata.getTable() )
 					&& !IgniteAssociationSnapshot.isThirdTableAssociation( associationKeyMetadata ) ) {
 				appendIndex( queryEntity, associationKeyMetadata, context );
 			}
 		}
-		result.setQueryEntities( Arrays.asList( queryEntity ) );
-		return result;
+		log.debugf( "queryEntity: %s", queryEntity );
+		cacheConfiguration.setQueryEntities( Arrays.asList( queryEntity ) );
+
+		return cacheConfiguration;
+	}
+
+	@SuppressWarnings("unchecked")
+	private void addTableInfo(QueryEntity queryEntity, SchemaDefinitionContext context, String tableName) {
+		Namespace namespace = context.getDatabase().getDefaultNamespace();
+		Optional<Table> tableOptional = namespace.getTables().stream().filter( currentTable -> currentTable.getName().equals( tableName ) ).findFirst();
+		if ( tableOptional.isPresent() ) {
+			Table table = tableOptional.get();
+			for ( Iterator<Column> columnIterator = table.getColumnIterator(); columnIterator.hasNext(); ) {
+				Column currentColumn = columnIterator.next();
+				Type valueType = currentColumn.getValue().getType();
+				Value value = currentColumn.getValue();
+				if ( valueType instanceof CustomType ) {
+					CustomType type = (CustomType) currentColumn.getValue().getType();
+					UserType userType = type.getUserType();
+					if ( userType instanceof EnumType ) {
+						EnumType enumType = (EnumType) type.getUserType();
+						if ( enumType.isOrdinal() ) {
+							queryEntity.addQueryField( currentColumn.getName(), Integer.class.getName(), null );
+						}
+						else {
+							queryEntity.addQueryField( currentColumn.getName(), String.class.getName(), null );
+						}
+					}
+				}
+				else if ( valueType instanceof YesNoType ) {
+					queryEntity.addQueryField( currentColumn.getName(), String.class.getName(), null );
+				}
+				else if ( value.getClass() == SimpleValue.class ) {
+					// it is simple type. add the field
+					SimpleValue simpleValue = (SimpleValue) value;
+					Class returnValue = simpleValue.getType().getReturnedClass();
+					returnValue = h2TypeMapping.getOrDefault( returnValue, returnValue );
+					queryEntity.addQueryField( currentColumn.getName(), returnValue.getName(), null );
```

Comment:
I thinnk all this part can be removed by using something like:

                TypeTranslator translator = serviceRegistry.getService( TypeTranslator.class );
		Type valueType = currentColumn.getValue().getType();
		GridType gridType = translator.getType( valueType )

But I'm a bit tired, I will continue tomorrow

Reply:
This solution doesn't seem to work, I'll play a bit more. My concern is that the logic here is  not taking into account how the dialect can change the way values are stored